packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -835,6 +835,18 @@ async function formNestedStack(
   const { amplifyMeta } = projectDetails;
   let authResourceName: string;
 
+  // Add CLI versioning information to the root stack's metadata
+  const metadata = nestedStack.Metadata || {};
+
+  Object.assign(metadata, {
+    AmplifyCLI: {
+      DeployedByCLIVersion: context.versionInfo.currentCLIVersion,
+      MinimumCompatibleCLIVersion: context.versionInfo.minimumCLIVersion,
+    },
+  });
+
+  nestedStack.Metadata = metadata;
```

Comment:
Can we put this logic and the similar logic in initializer.ts in a function like `addCLIVersionInfoToStackMeta`? This function should contain logic for not overwriting existing meta if present which will allow us to easily add other meta in the future.

Reply:
initializer already has logic for this, push resources is where we update with new information if needed and as we don't have any other meta now, I'd not move this, the whole push needs a per category rewrite soon.